browser/src/react-redux/App/LoginPage.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import logInUser from './logInUser.action'
+import axios from 'axios'
+
+const initialState = {
+  email: '',
+  password: '',
+  generalError: ''
+}
+
+class LoginPage extends React.Component {
+  state = initialState
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+  }
+
+  onSubmit = e => {
+    this.setState(initialState)
+
+    e.preventDefault()
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    }
+
+    axios
+      .post('/api/auth/login', user)
+      .then(res => {
+        this.props.logInUser(res.data.user)
+        //if login successfull reroute to home
```

Comment:
We should probably check that the redux store has actually been updated instead of assuming. Remember, if this is successful, we will eventually receive the user through props, because you connected this component with the store.

I also noticed that, even though you connected the user, you're not doing anything with. You'll normally avoid subscribing a component to store changes if you don't care about those changes. _Luckily_, because of the requested change that I mentioned above, we now do care! In the `.then()`, let's just worry about calling the action. Then, in a lifecycle method or using a redirect in the render, we can simply redirect the user when we finally detect that `this.props.user` is populated.

Reply:
I actually just created a ticket for the signup form to do this as well, but while we're changing this in the login form, let's just _always_ redirect the user if they are already logged in - no matter if they just logged in through this view, or were already logged in before they got to this view.